src/Microsoft.ML.OnnxTransformer/OnnxUtils.cs

```diff
@@ -439,7 +441,15 @@ public static NamedOnnxValue CreateScalarNamedOnnxValue<T>(string name, T data)
         {
             if (!_onnxTypeMap.Contains(typeof(T)))
                 throw new NotImplementedException($"Not implemented type {typeof(T)}");
-            return NamedOnnxValue.CreateFromTensor<T>(name, new DenseTensor<T>(new T[] { data }, new int[] { 1 }));
+
+            if (data.GetType() == typeof(ReadOnlyMemory<Char>))
```

Comment:
Should be `if (typeof(T) == typeof(ReadOnlyMemory<Char>))`. This allows the compiler/runtime to remove this check when `typeof(T)` is `int`, `float`, etc.